test/runtests.jl

```diff
@@ -1446,3 +1564,148 @@ end
+                    -3.3333334972350723 -3.3333334972350723 -0.0 -3.3333334972350723 -0.0; 
+                    4.333333497138949 0.0 4.333333497138949 4.333333497138949 0.0; 
+                    4.000000005907649 0.0 0.0 0.0 4.000000005907649; 
+                    -0.33333334610634496 -0.33333334610634496 -0.0 -0.0 -0.33333334610634496; 
+                    -3.6666667654825043 -0.0 -3.6666667654825043 -0.0 -3.6666667654825043]
+        
+        gm_poisw = fit(GeneralizedLinearModel, f, dobson, Poisson(), wts = dobson.Weights)
+
+        mm0_poisw = [-0.9624647521850039 -0.0 -0.0 -0.0 -0.0; 
+                      0.6901050904949885 0.6901050904949885 0.0 0.0 0.0; 
+                      0.2723596655008255 0.0 0.2723596655008255 0.0 0.0; 
+                     -0.9062167634177802 -0.0 -0.0 -0.9062167634177802 -0.0; 
+                     -0.7002548908882033 -0.7002548908882033 -0.0 -0.7002548908882033 -0.0; 
+                      1.606471661159352 0.0 1.606471661159352 1.606471661159352 0.0; 
+                      1.8686815106332157 0.0 0.0 0.0 1.8686815106332157; 
+                      0.010149793505874801 0.010149793505874801 0.0 0.0 0.010149793505874801; 
+                     -1.8788313148033928 -0.0 -1.8788313148033928 -0.0 -1.8788313148033928]
+        
+        
+        
+        @test mm0_pois ≈  GLM.momentmatrix(gm_pois) atol=1e-06
+        @test mm0_poisw ≈  GLM.momentmatrix(gm_poisw) atol=1e-06
+    end
+    @testset "Binomial" begin
+        f = @formula(admit ~ 1 + rank)
+        
+        gm_bin = fit(GeneralizedLinearModel, f, admit_agr, Binomial())
+        mm0_bin = [-0.5  -0.0  -0.0  -0.0
+                   -0.5  -0.5  -0.0  -0.0
+                   -0.5  -0.0  -0.5  -0.0
+                   -0.5  -0.0  -0.0  -0.5
+                    0.5   0.0   0.0   0.0
+                    0.5   0.5   0.0   0.0
+                    0.5   0.0   0.5   0.0
+                    0.5   0.0   0.0   0.5]        
+        @test mm0_bin ≈ GLM.momentmatrix(gm_bin)     
+
+        gm_binw = fit(GeneralizedLinearModel, f, admit_agr, Binomial(), wts=aweights(admit_agr.count))
+        mm0_binw =  [-15.1475    -0.0      -0.0     -0.0
+                     -34.6887   -34.6887   -0.0     -0.0
+                     -21.5207    -0.0     -21.5207  -0.0
+                      -9.85075   -0.0      -0.0     -9.85075
+                      15.1475     0.0       0.0      0.0
+                      34.6887    34.6887    0.0      0.0
+                      21.5207     0.0      21.5207   0.0
+                       9.85075    0.0       0.0      9.85075]
+        
+        @test mm0_binw ≈ GLM.momentmatrix(gm_binw) atol=1e-03
+        Vcov =[ 0.0660173  -0.0660173  -0.0660173  -0.0660173
+               -0.0660173   0.0948451   0.0660173   0.0660173
+               -0.0660173   0.0660173   0.112484    0.0660173
+               -0.0660173   0.0660173   0.0660173   0.167532]
+
+        ## This is due to divverences between chol and qr
```

Comment:
Where is QR used? Differences large enough to require a 1e-3 tolerance are a bit annoying. Wouldn't using a smaller tolerance when fitting the model allow getting closer values?
```suggestion
        ## This is due to differences between chol and qr
```